### PR TITLE
Fast-forward to the final v1 release:

### PR DESCRIPTION
**Pull request by glima**
The versioning was finally fixed to the one we're going to following
from the release on (similar to kmod's and systemd's).

All the bundled software now comes together with github's soletta
tarball release and we don't have to merge them back in the source code
for compiling anymore.

All nodes marked as "default n" on Soletta _won't_ be packaged for this
release. It may change on future releases only.

Since Fedora 22 was EOL-ed, we now check for libmicrohttpd's version
from 24 onwards (we also bump the required version to what should have
been -- it did not make it to the v1 release, unfortunately). All tests
depending on it get disabled too, when applicable. linux-micro won't be
of concern anymore also, with no more Fedora 22.

All licences were carefully revised.

Changelog:
- mavlink support added
- 100% C API covered by docs
- many IIO node types added
- Reviewed structs for holes and checks for struct versions on exposed API
- Update and fix all board samples (Minnow, Edison and Galileo)
- Some fixes on I/O implementations
- Improvement on sockets API
- Added many IIO node samples
- Light sensor category added to Linux IIO
- Added support to STTS751 temperature sensor
- Basic infrastructure added for Node.js bindings
- LWM2M protocol support added
- Support I/O on Zephyr OS
- Increased amount of node types supported by GTK (simulation)
- Reduced memory consumption by CoAP - LWM2M samples are running fine on RIOT
- Single node support - so it’s possible to have a single node without
  an associated flow. Useful when you need to access a component, send
  packets to its input ports manually and be notified when it’s
  sending packets on its output ports.
- Fixed pin mux issue when using Edison without the arduino board
- Fixes on HTTP implementation regarding IPv6
- Many minor API changes (function / struct names to follow convention)
- RGB and direction-vector persistence nodes created
- Tutorial on docs
- Added support for secure OIC connections using pre shared certs
- Node.js OIC bindings implemented
- Fix OIC to deal with cases where IPv4 isn’t enabled
- Dependencies of samples were fixed
- Added web inspector to sol-fbp-runner
- GPIO, AIO, PWM and UART bindings for Node.js
- Added API to server side events on http-server with samples. It was
  supported on http-client as well.
- Many changes on LWM2M API in order to have a more consistent API for
  all communication protocols
- Provide node types for a lot of OIC resources data models
- Changes on CoAP implementation improving memory-efficiency
- Make OIC device IDs and resource structure compatible with IoTivity 1.1 RC3
- Add API to discover OIC resources using resource interface
- Many fixes on OIC generator
- Make IIO device creation/addressing synchronous
- Add simple JSON types as HTTP node types
- Add JS API for SPI, I2C
- Tons of new tests, samples and fixes
- LWM2M bootstrap interface implemented
- Connections management support (sol-netctl)
- Support HTTPS on http-server
- Introduced sol-bluetooth and sol-gatt API
- Support to node type aliases
- Many samples for Nodejs bindings were merged
- A couple tools were created to help debugging I/O (sol-aio and sol-gpio)
- Node types related to robotics were added
- Lots of changes on C API, leading to a more uniform interface across
  all different parts (I/O, communication protocols, ...)

Signed-off-by: Gustavo Lima Chaves gustavo.lima.chaves@intel.com

### CODE REVIEW

**Comment by glima** [comment]
@cabelitos , @bdilly  ^

File: rpm/soletta.spec [hunk 1]
```diff
@@ -1,46 +1,57 @@
...
-%global soletta_build 1
-%global soletta_tag beta13
+%global soletta_version 1
+%global soletta_current 1
+%global soletta_revision 0
+%global soletta_age 0
 
-%global soletta_duktape_version 1.2.2
-%global soletta_duktape_tag beta2
+%global soletta_major %(echo $(( %{soletta_current} - %{soletta_age} )))
```
Comment (bdilly) [inline rpm/soletta.spec:6]: not sure if it's the way to do it on a spec file, but I assume you verified already

**Comment by bdilly** [comment]
:+1:

File: rpm/soletta.spec (hunk 1 above)
Comment (glima) [inline rpm/soletta.spec:6]: You mean syntactically? It works. It also matches what we want: rpmbuild output is "build/soletta_sysroot/usr/lib64/libsoletta.so.1.0.0"

**Pull request merged**